src/components/shared/data-set-view-button.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useContext } from "react";
+import { getTileComponentInfo } from "../../models/tiles/tile-component-info";
+import { SharedDataSet } from "../../models/shared/shared-data-set";
+import { AddTilesContext, TileModelContext } from "../tiles/tile-api";
+import { TileToolbarButton } from "./tile-toolbar-button";
+
+import ViewBadgeIcon from "../../assets/icons/view/view-badge.svg";
+
+import "./data-set-view-button.scss";
+
+interface IProps {
+  newTileType: string;
+}
+
+export const DataSetViewButton: React.FC<IProps> = ({newTileType}) => {
+  const addTilesContext = useContext(AddTilesContext);
+  const tile = useContext(TileModelContext);
+
+  // TODO: if the document or tile are undefined then disable the button
```

Comment:
The case I was thinking about was is if the toolbar is somehow shown on a curriculum document. I think in that case addTilesContext will be undefined. It should just be theoretical though since tiles shouldn't be showing their toolbars when read-only. But since this depends on the tile developer doing the right thing it might happen in some cases.